language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/workspace/BallerinaWorkspaceManagerProxy.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ballerinalang.langserver.workspace;
+
+import io.ballerina.projects.Project;
+import org.ballerinalang.langserver.LSContextOperation;
+import org.ballerinalang.langserver.common.utils.CommonUtil;
+import org.ballerinalang.langserver.commons.LanguageServerContext;
+import org.ballerinalang.langserver.commons.workspace.WorkspaceDocumentException;
+import org.ballerinalang.langserver.commons.workspace.WorkspaceManager;
+import org.eclipse.lsp4j.DidChangeTextDocumentParams;
+import org.eclipse.lsp4j.DidCloseTextDocumentParams;
+import org.eclipse.lsp4j.DidOpenTextDocumentParams;
+
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.Locale;
+import java.util.Optional;
+
+/**
+ * Ballerina workspace manager proxy implementation.
+ * This proxy maintains two workspace managers, one for the expr file scheme based documents and the default manager
+ * for the file scheme based documents.
+ *
+ * @since 2.0.0
+ */
+public class BallerinaWorkspaceManagerProxy implements WorkspaceManagerProxy {
+    private static final String EXPR_SCHEME = "expr";
```

Comment:
We have this constant defined in CommonUtil as well.

Reply:
Done